src/instructlab/utils.py

```diff
@@ -215,18 +215,19 @@ def get_documents(
                 temp_dir=temp_dir,
                 skip_checkout=skip_checkout,
             )
-            file_contents = []
 
             logger.debug("Processing files...")
+            opened_files = False
             for pattern in file_patterns:
                 for file_path in glob.glob(os.path.join(repo.working_dir, pattern)):
                     if os.path.isfile(file_path) and file_path.endswith(".md"):
-                        with open(file_path, "r", encoding="utf-8") as file:
-                            file_contents.append(file.read())
+                        with open(file_path, "r", encoding="utf-8"):
+                            # Success! we could open the file.
+                            # We don't actually care about the contents, though.
+                            opened_files = True
```

Comment:
This check is slightly different. The prior code tests if there is any content in the aggregated files. If all the files are empty, then an exception is raised. Now we just check that the files can be opened. 


```suggestion
                        if os.path.isfile(file_path) and os.path.getsize(file_path) > 0:
    opened_files = True
                            # Success! The file in non-empty.
                            # We don't actually care about the contents, though.
                            opened_files = True
```

This avoids opening the file, which may be a time saver too.

Reply:
Are you sure? I think `file.read() will return '' on empty files, so the code is equivalent - it will **not** raise when all files are empty.